src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -26,13 +26,24 @@ internal class RpcWorkerConfigFactory
         private Dictionary<string, RpcWorkerDescription> _workerDescripionDictionary = new Dictionary<string, RpcWorkerDescription>();
 
         public RpcWorkerConfigFactory(IConfiguration config, ILogger logger, ISystemRuntimeInformation systemRuntimeInfo, IEnvironment environment, IMetricsLogger metricsLogger)
+            : this(config, logger, systemRuntimeInfo, environment, metricsLogger, Directory.Exists)
+        {
+        }
+
+        internal RpcWorkerConfigFactory(IConfiguration config, ILogger logger, ISystemRuntimeInformation systemRuntimeInfo, IEnvironment environment, IMetricsLogger metricsLogger, Func<string, bool> directoryExists)
```

Comment:
Rather than exposing `Directory.Exists` like this -- can you make an internal static method on here like `GetWorkersDirPath`  that has all this logic (and it has this Func<> on it)? Then you can test that method directly. That'd keep it out of the constructor and restricts it's use to the one place it's needed.